diff -up kmod-25/libkmod/libkmod.h.omv~ kmod-25/libkmod/libkmod.h
--- kmod-25/libkmod/libkmod.h.omv~	2018-07-26 00:10:32.258307004 +0200
+++ kmod-25/libkmod/libkmod.h	2018-07-26 00:11:02.277308648 +0200
@@ -71,8 +71,10 @@ enum kmod_index {
 	KMOD_INDEX_MODULES_ALIAS,
 	KMOD_INDEX_MODULES_SYMBOL,
 	KMOD_INDEX_MODULES_BUILTIN,
-	/* Padding to make sure enum is not mapped to char */
-	_KMOD_INDEX_PAD = (1 << 31),
+	/* Padding to make sure enum is not mapped to char,
+	 * but still evaluates to a positive number in a
+	 * signed 32-bit data type */
+	_KMOD_INDEX_PAD = (1 << 30),
 };
 int kmod_dump_index(struct kmod_ctx *ctx, enum kmod_index type, int fd);
 
