diff -Naur kmod-26/configure.ac kmod-26.tpg/configure.ac
--- kmod-26/configure.ac	2019-11-28 11:55:53.595047847 +0000
+++ kmod-26.tpg/configure.ac	2019-11-28 11:56:37.741048101 +0000
@@ -18,7 +18,7 @@
 AM_SILENT_RULES([yes])
 LT_INIT([disable-static pic-only])
 
-AS_IF([test "x$enable_static" = "xyes"], [AC_MSG_ERROR([--enable-static is not supported by kmod])])
+dnl AS_IF([test "x$enable_static" = "xyes"], [AC_MSG_ERROR([--enable-static is not supported by kmod])])
 AS_IF([test "x$enable_largefile" = "xno"], [AC_MSG_ERROR([--disable-largefile is not supported by kmod])])
 
 #####################################################################
