--- kmod-20/configure.ac.0000~	2015-03-04 14:58:02.326789903 +0100
+++ kmod-20/configure.ac	2015-03-04 14:58:45.356403755 +0100
@@ -18,7 +18,7 @@ AM_INIT_AUTOMAKE([check-news foreign 1.1
 AM_SILENT_RULES([yes])
 LT_INIT([disable-static pic-only])
 
-AS_IF([test "x$enable_static" = "xyes"], [AC_MSG_ERROR([--enable-static is not supported by kmod])])
+dnl AS_IF([test "x$enable_static" = "xyes"], [AC_MSG_ERROR([--enable-static is not supported by kmod])])
 AS_IF([test "x$enable_largefile" = "xno"], [AC_MSG_ERROR([--disable-largefile is not supported by kmod])])
 
 #####################################################################
